extensions/amp-apester-media/0.1/amp-apester-media.js

```diff
@@ -142,18 +200,41 @@ class AmpApesterMedia extends AMP.BaseElement {
    * @return {string}
    * */
   constructUrlFromMedia_(id) {
-    return `${this.rendererBaseUrl_}/interaction/${encodeURIComponent(id)}`;
+    const conditionalChannelId = this.embedOptions_distributionChannelId ?
+      `&channelId=${this.embedOptions_distributionChannelId}` : '';
+    const conditionalInteractionType =
+      `&type=${encodeURIComponent(this.embedOptions_.playlist ? 'playlist' :
+        'editorial')}`;
+    const platform = `&platform=${getPlatform()}`;
+    const cannonicalUrl = document.querySelector('link[rel=\'canonical\']')
+      ? document.querySelector('link[rel=\'canonical\']').getAttribute('href')
+      : document.location.href;
+    return `${this.rendererBaseUrl_}/interaction/${
+      encodeURIComponent(id)}?sdk=amp&canonicalUrl=${
+      encodeURIComponent(cannonicalUrl)}${
+      conditionalChannelId}${
+      conditionalInteractionType}${
+      platform}`;
   }
 
-  /** @param {string} src
-   * @return {!Element}
+  /**
+   * Constructs the media Iframe.
+   * @param src
+   * @param mediaId
+   * @private
    */
-  constructIframe_(src) {
+  constructIframe_(src, media) {
     const iframe = this.element.ownerDocument.createElement('iframe');
     iframe.setAttribute('frameborder', '0');
     iframe.setAttribute('allowtransparency', 'true');
     iframe.setAttribute('scrolling', 'no');
-    iframe.src = src;
+    iframe.setAttribute('data-interaction-id', media.interactionId);
+    iframe.name = /wpcomwidgets/.test(document.location.href) ?
```

Comment:
can you explain what's happening here? the href would be the AMP page using the component and in post cases an AMP CDN Cahe URL, what does the check for `wpcomwidgets` supposed to do? Also use `this.win.location.href` instead of `document.location`

Reply:
Well, we sometimes get embedded in a WP.com widget iframe. In that case we are using the referrer in order to understand where is the current media unit lives. 
But, after rethinking I guess that it won't be happening with an AMP page so I will remove it.